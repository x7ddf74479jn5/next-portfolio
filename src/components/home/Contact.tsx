import { NextArrowButton } from "src/components/common/NextArrowButton";
import Spacer from "src/components/layouts/Spacer";

const Contact = () => {
  return (
    <section className="c-section c-bg-primary">
      <div className="c-section-wrapin p-grid__contact">
        <h2>お仕事のご依頼・ご相談</h2>
        <hr className="c-section__line" />
        <Spacer size="md" />
        <p>
          企業HPなどのWebサイト制作、チャットボット開発、WordPress導入・カスタマイズなどのお見積もり、ご相談はお気軽にご連絡ください。
          お客様だけでなく、エンドユーザーにも喜んでいただけるよう技術を提供します。
        </p>
        <Spacer size="sm" />
        <NextArrowButton label="お問い合わせ" href="/contact/" />
        <Spacer size="md" />
      </div>
    </section>
  );
};

export default Contact;
