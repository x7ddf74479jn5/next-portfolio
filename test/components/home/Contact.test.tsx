import Contact from "src/components/home/Contact";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Contact", () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Contact />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("heading")).toHaveTextContent("お仕事");
    expect(renderResult.getByRole("separator")).toBeInTheDocument();
    expect(renderResult.getByRole("button")).toHaveTextContent("お問い合わせ");
    expect(renderResult.getByRole("button")).toHaveAttribute("href", "/contact/");
  });
});
