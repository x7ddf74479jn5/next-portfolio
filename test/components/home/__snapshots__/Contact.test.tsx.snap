// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgPrimary"
  >
    <div
      class="sectionWrapIn"
    >
      <div
        class="gridContact"
      >
        <h2>
          お仕事のご依頼・ご相談
        </h2>
        <hr
          class="sectionLine"
        />
        <div
          class="md"
        />
        <p>
          企業HPなどのWebサイト制作、チャットボット開発、WordPress導入・カスタマイズなどのお見積もり、ご相談はお気軽にご連絡ください。 お客様だけでなく、エンドユーザーにも喜んでいただけるよう技術を提供します。
        </p>
        <div
          class="sm"
        />
        <a
          class="btnRound iconArrowNext"
          href="/contact"
          role="button"
        >
          お問い合わせ
        </a>
        <div
          class="md"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
